.github/workflows/compilation_on_android_ubuntu.yml

```diff
@@ -431,13 +436,13 @@ jobs:
         run: |
           cd samples/wasi-threads
           mkdir build && cd build
-          cmake -DWASI_SYSROOT=`pwd`/../../../core/deps/wasi-libc/sysroot ..
+          cmake ${{ matrix.wasi_sysroot_option }} ..
           cmake --build . --config Release --parallel 4
           ./iwasm wasm-apps/no_pthread.wasm
 
   test:
     needs: [build_iwasm, build_llvm_libraries_on_ubuntu_2004, build_wamrc]
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```

Comment:
It may be a risk to upgrade the default test OS to Ubuntu-22.04. Maybe we can keep this PR open, and wait until the wasi-sdk-20.0 for threads works for Ubuntu-20.04?

Reply:
No strong opinion, let's wait for the decision on the issue I opened. Or, as an alternative, we could run those tests both with 20.04 and 22.04.

Reply:
What's the risk? In https://github.com/WebAssembly/wasi-sdk/issues/303, I mentioned that I wouldn't mind if someone else (@eloparco?) works on getting wasi-sdk to build on older machines but I wasn't personally too interested. If this workflow is the only reason to do so, I wonder if we could resolve whatever risks are here? I assume this workflow would have to be updated at some point anyways since I wouldn't assume it to pin to `20.04` forever.

Reply:
I was just worried about the spec cases and wasi-threads cases are not tested on Ubuntu-20.04, and we don't know whether there are errors reported on it. I discussed with @lum1n0us, he also agreed to run these tests on both Ubuntu-20.04 and Ubuntu-22.04, it seems better. So @eloparco, could you help enable that?